plugins/redmine_bugs_ruby_lang/lib/redmine_mailing_list_integration/redmine_ext/mailer.rb

```diff
@@ -1,74 +1,51 @@
+    super
+  end
 
-    m = super(user, issue)
+  private
 
-    m.header[:to] = mailing_lists.map(&:address)
-    m.header[:subject] = "[#{issue.project.name} #{issue.tracker.name}##{issue.id}] #{issue.subject}"
-    m
+  def deliver_issue_add_to_mailing_lists(issue)
+    mail = issue_add(issue.author, issue)
+    mailing_lists = issue.project.mail_routes_for_issue(issue)
+    deliver_to_mailing_lists(mail, mailing_lists, issue: issue)
   end
 
-  def issue_edit(user, journal)
+  def deliver_issue_edit_to_mailing_lists(journal)
     issue = journal.issue
+    mail = issue_edit(issue.author, journal)
     mailing_lists = issue.project.mail_routes_for_issue(issue)
-
-    record_message(issue, journal, mailing_lists)
-
-    m = super(user, journal)
-
-    m.header[:to] = mailing_lists.map(&:address)
-    m.header[:subject] = "[#{issue.project.name} #{issue.tracker.name}##{issue.id}] #{issue.subject}"
-    m
+    deliver_to_mailing_lists(mail, mailing_lists, issue: issue, journal: journal)
   end
 
-  def attachments_added(user, attachments)
-    mailing_lists = attatchments.first.container.project.mail_routes_for_attachments(attachments)
-
-    m = super(user, attachments)
-
-    m.header[:to] = mailing_lists.map(&:address)
-    m
+  def deliver_attachments_added_to_mailing_lists(attachments)
+    container = attachments.first.container
+    mail = attachments_added(container.author, attachments)
+    mailing_lists = container.project.mail_routes_for_attachments(attachments)
+    deliver_to_mailing_lists(mail, mailing_lists)
   end
 
-  private
-
-  def record_message(issue, journal, mailing_lists)
-    message_record_ids = mailing_lists.map {|ml|
-      record = MailingListMessage.create!(
-        mailing_list: ml,
-        issue: issue,
-        journal: journal
-      )
-      record.id
-    }
-    headers['X-Redmine-MailingListIntegration-Message-Ids'] = message_record_ids.join(",")
-  end
-end
-
-Mailer.class_eval do
-  class << self
-    def deliver_issue_add(issue)
-      unless issue.originates_from_mail?
-        issue_add(issue.author, issue).deliver_later
+  def deliver_to_mailing_lists(mail, mailing_lists, record_attributes = nil)
+    if record_attributes
+      records = mailing_lists.map do |mailing_list|
+        MailingListMessage.create!(mailing_list: mailing_list, **record_attributes)
       end
+      mail.header["X-Redmine-MailingListIntegration-Message-Ids"] = records.map(&:id).join(",")
     end
-
-    def deliver_issue_edit(journal)
-      unless journal.originates_from_mail?
-        issue_edit(journal.issue.author, journal).deliver_later
-      end
-    end
-
-    def deliver_attachments_added(attachments)
-      attachment = attachments.first
-      return unless attachment.container_type == 'Issue'
-      attachments_added(attachment.container.author, attachments).deliver_later
-    end
+    mail.to = mailing_lists.map(&:address)
```

Comment:
RuntimeError: You've accessed the message before asking to deliver it later, so you may have made local changes that would be silently lost if we enqueued a job to deliver it. Why? Only the mailer method *arguments* are passed with the delivery job! Do not access the message in any way if you mean to deliver it later. Workarounds: 1. don't touch the message before calling #deliver_later, 2. only touch the message *within your mailer method*, or 3. use a custom Active Job instead of #deliver_later.

Reply:
@jonathanhefner thanks for looking into this. However, this attempt to simplify things doesn't seem to work correctly.

Reply:
@ahorek Thank you for testing! :pray:  I hadn't thought of that.  I've pushed a new commit that should avoid that problem.